src/components/TweetCreationForm.tsx

```diff
@@ -62,25 +69,41 @@ const TweetCreationForm: React.FC = () => {
     "translation:tweetCreationForm.whoCanAnswer.button.all"
   );
 
+  const urlSearch = new RegExp(
+    /https?:\/\/(?:www.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9].[^\s<]{2,}|https?:\/\/(?:www.|(?!www))[a-zA-Z0-9]+.[^\s<]{2,}/,
+    "g"
+  );
+
   const handleTweetButton = (): void => {
-    dispatch(tweetsActions.createTweet({ content: content }));
+    const tweetContent = content.replace(
+      urlSearch,
+      (result) => `<a href="${result}">${result}</a>`
+    );
```

Comment:
we don't need to detect links at this step anymore, and instead this functionality should be only in place of render tweet